frontend/views/containers/proposals/ProposalBox.vue

```diff
@@ -65,9 +77,28 @@ export default {
         year: 'numeric', month: 'long', day: 'numeric'
       })
     },
+    shouldTruncateReason () {
+      const reason = this.proposal.data.proposalData.reason
+      const threshold = 40 // avoid clicking "read more" and see only a few more characters.
+      return reason.length > this.config.reasonMaxLength + threshold
+    },
     humanReason () {
       const reason = this.proposal.data.proposalData.reason
-      return reason ? `"${reason}"` : undefined
+      const maxlength = this.config.reasonMaxLength
+      if (this.ephemeral.isReasonHidden && this.shouldTruncateReason) {
+        // Prevent "..." to be added after an empty space. ex: "they would ..." -> "they would..."
+        const charToTruncate = reason.charAt(maxlength - 1) === ' ' ? maxlength - 1 : maxlength
+        return `"${reason.substr(0, charToTruncate)}..."`
+      }
+
+      this.ephemeral.isReasonHidden = false // eslint-disable-line vue/no-side-effects-in-computed-properties
```

Comment:
There should be no side-effects if you are not aware of them or don't handle them correctly. In this case, this side-effect is doing nothing wrong... but I'll look again and try to avoid it